### ISSUE TITLE
Error while loading page which fetches data from an API call

### ISSUE THREAD

**Issue by Satish-Lakhani** [arunoda/learnnextjs-demo#24]
Hi, 
I'm new to react (coming from angular side), and totally new to SSR concept.

As per you code below,
```
import Layout from '../components/Layout.js'
import fetch from 'isomorphic-unfetch'

const Post =  (props) => (
    <Layout>
       <h1>{props.show.name}</h1>
       <p>{props.show.summary.replace(/<[/]?p>/g, '')}</p>
       <img src={props.show.image.medium}/>
    </Layout>
)

Post.getInitialProps = async function (context) {
  const { id } = context.query
  const res = await fetch(`https://api.tvmaze.com/shows/${id}`)
  const show = await res.json()

  console.log(`Fetched show: ${show.name}`)

  return { show }
}

export default Post
```
This is running fine on client side, but while running on server side (reload detail page), it throws below error.
![selection_011](https://user-images.githubusercontent.com/9801442/39312868-2fa695be-498e-11e8-902b-91d7860ef508.png)

It seems like page gets rendered before data from API is available.
What would be the best way to handle such a situations?
Is it like showing or returning 'Loading' message or loading component till data gets available? If yes then mostly in all cases where we fetch data using API call which takes some time, it will always return rendered component with 'Loading' only.

**Comment by ksuzushima** [comment]
@Satish-Lakhani You might have already resolved but,

`Then restart your app to apply the above code change.`
if you do not restart, context object still have `title` query param and `id` is undefined.

**Comment by qWici** [comment]
@ksuzushima yep.I have the same error and restart server solved this problem. You can close this issue)
